libr/main/rasm2.c

```diff
@@ -34,6 +34,9 @@ static RAsmState *__as_new() {
 	if (as) {
 		as->l = r_lib_new (NULL, NULL);
 		as->a = r_asm_new ();
+		if (as->a) {
```

Comment:
This is wrong. Becasue core creates the instance, and owns it. So the num instance you create here is leaking and lost.

Reply:
We are not passing core into `r_main_rasm2()`. And asan shows no relevant leaks.

Reply:
Im talking about r2. I know this is fine for rasm2

Reply:
r2 doesn't call that function. `__as_new()` is a static function only visible in `libr/main/rasm2.c` and only called from `r_main_rasm2()`.